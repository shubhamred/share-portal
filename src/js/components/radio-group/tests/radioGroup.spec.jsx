/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import RadioGroup from '../radioGroup';

configure({ adapter: new Adapter() });

describe('components/RadioGroup', () => {
  it('Renders default options', () => {
    const input = {
      onChange: jest.fn()
    };
    const options = ['ABC', '123'];
    const component = shallow(
      <RadioGroup input={input} options={options} />
    );
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });
  it('Renders disables options', () => {
    const input = {
      onChange: jest.fn()
    };
    const options = ['ABC', '123'];
    const component = shallow(
      <RadioGroup input={input} options={options} disabled={true} />
    );
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });
  it('Renders with options', () => {
    const props = {
      input: {
        onChange: jest.fn(),
        value: ''
      },
      options: ['ABC', '123'],
      selected: 'ABC',
      label: 'Label',
      hideButton: true
    };
    const component = shallow(
      <RadioGroup {...props} />
    );
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });
  it('Renders with options', () => {
    const props = {
      input: {
        onChange: jest.fn(),
        value: 'ABC'
      },
      options: ['ABC', '123'],
      selected: 'ABC',
      label: 'Label',
      hideButton: true,
      onHideBtnClick: jest.fn()
    };
    const component = mount(
      <RadioGroup {...props} />
    );
    component.find('.hideBtn').simulate('click');
    expect(props.onHideBtnClick).toBeCalled();
  });
  it('Handles On change', () => {
    const props = {
      input: {
        onChange: jest.fn(),
        value: 'ABC'
      },
      options: ['ABC', '123'],
      selected: 'ABC',
      label: 'Label',
      hideButton: true,
      onHideBtnClick: jest.fn(),
      handleChange: jest.fn()
    };
    const component = mount(
      <RadioGroup {...props} />
    );
    component.find('input[value="123"]').simulate('change');
    expect(props.input.onChange).toBeCalled();
    expect(props.handleChange).toBeCalled();
  });
});
