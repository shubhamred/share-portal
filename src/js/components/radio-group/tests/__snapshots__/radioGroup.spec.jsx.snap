// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/RadioGroup Renders default options 1`] = `
<div>
  <WithStyles(ForwardRef(FormControl))
    component="fieldset"
    disabled={false}
  >
    <WithStyles(ForwardRef(FormLabel))
      className="label"
    />
    <ForwardRef(RadioGroup)
      className="radioGroup"
      onChange={[Function]}
    >
      <WithStyles(ForwardRef(FormControlLabel))
        className="input"
        control={<ForwardRef(WithStyles(Component)) />}
        key="option-ABC"
        label="ABC"
        value="ABC"
      />
      <WithStyles(ForwardRef(FormControlLabel))
        className="input"
        control={<ForwardRef(WithStyles(Component)) />}
        key="option-123"
        label="123"
        value="123"
      />
    </ForwardRef(RadioGroup)>
  </WithStyles(ForwardRef(FormControl))>
</div>
`;

exports[`components/RadioGroup Renders disables options 1`] = `
<div>
  <WithStyles(ForwardRef(FormControl))
    component="fieldset"
    disabled={true}
  >
    <WithStyles(ForwardRef(FormLabel))
      className="label"
    />
    <ForwardRef(RadioGroup)
      className="radioGroup"
      onChange={[Function]}
    >
      <WithStyles(ForwardRef(FormControlLabel))
        className="input"
        control={<ForwardRef(WithStyles(Component)) />}
        key="option-ABC"
        label="ABC"
        value="ABC"
      />
      <WithStyles(ForwardRef(FormControlLabel))
        className="input"
        control={<ForwardRef(WithStyles(Component)) />}
        key="option-123"
        label="123"
        value="123"
      />
    </ForwardRef(RadioGroup)>
  </WithStyles(ForwardRef(FormControl))>
</div>
`;

exports[`components/RadioGroup Renders with options 1`] = `
<div>
  <WithStyles(ForwardRef(FormControl))
    component="fieldset"
    disabled={false}
  >
    <WithStyles(ForwardRef(FormLabel))
      className="label"
    >
      Label
      <HideButton
        isSection={false}
        onHideBtnClick={[Function]}
        visible={false}
      />
    </WithStyles(ForwardRef(FormLabel))>
    <ForwardRef(RadioGroup)
      className="radioGroup"
      onChange={[Function]}
      value="ABC"
    >
      <WithStyles(ForwardRef(FormControlLabel))
        className="input"
        control={<ForwardRef(WithStyles(Component)) />}
        key="option-ABC"
        label="ABC"
        value="ABC"
      />
      <WithStyles(ForwardRef(FormControlLabel))
        className="input"
        control={<ForwardRef(WithStyles(Component)) />}
        key="option-123"
        label="123"
        value="123"
      />
    </ForwardRef(RadioGroup)>
  </WithStyles(ForwardRef(FormControl))>
</div>
`;
